clash-vexriscv/src/VexRiscv.hs

```diff
@@ -505,23 +510,28 @@ vexRiscv# !_sourcePath clk rst0
 
 -- | Return a function that performs an execution step and a function to free
 -- the internal CPU state
-vexCPU :: IO
-  ( Ptr VexRiscv
-  , Ptr VexRiscv -> NON_COMB_INPUT -> IO OUTPUT           -- initStage1
-  , Ptr VexRiscv -> COMB_INPUT -> IO ()                   -- initStage2
-  , Ptr VexRiscv -> Word64 -> NON_COMB_INPUT -> IO OUTPUT -- rising
-  , Ptr VexRiscv -> Word64 -> COMB_INPUT -> IO ()         -- falling
-  , Ptr VexRiscv -> IO ()
-  )
-vexCPU = do
+vexCPU ::
+  -- | VCD dump path
+  Maybe FilePath ->
+  IO
+    ( Ptr VexRiscv
+    , Ptr VexRiscv -> NON_COMB_INPUT -> IO OUTPUT           -- initStage1
+    , Ptr VexRiscv -> COMB_INPUT -> IO ()                   -- initStage2
+    , Ptr VexRiscv -> Word64 -> NON_COMB_INPUT -> IO OUTPUT -- rising
+    , Ptr VexRiscv -> Word64 -> COMB_INPUT -> IO ()         -- falling
+    , Ptr VexRiscv -> IO ()
+    )
+vexCPU maybeVcdPath = do
   v <- vexrInit
+  vcdPath <- maybe (pure nullPtr) newCString maybeVcdPath
+  vcd <- vexrInitVcd v vcdPath
```

Comment:
Would it make sense to make the call of that function dependent on whether a path is set?

```suggestion
  vcd <- case maybeVcdPath of
     Nothing -> pure nullPtr
     Just path -> do
       vcdPath <- newCString path
       vexrInitVcd v vcdPath
```

Reply:
Ah yes, that's better